### PR TITLE
Add functionality to the renaming suggestions in PR [#2262]

### PR DESCRIPTION
**Pull request by iamalwaysuncomfortable**
## Motivation

PR #2262 suggests that documentation be updated to be more clear. This PR adds the suggested documentation into the CLI tool.

## Changelog
* Refactors the node configuration options as flags
* Adds a separate command line option for --private_key to disambiguate where the private key should be specified
* Adds renaming suggestions in the documentation & cli suggested in #2262 

## Test Plan
* All unit tests currently pass, added a test to ensure the tool is not allowed to start with conflicting `node type` flags
* If these changes are philosophically accepted, further test coverage will be written to cover further edge cases of the start command

## Related PRs
#2262

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -198,11 +200,11 @@ It is important to initialize the nodes starting from `0` and incrementing by `1
 
 The following is a list of options to initialize a node (replace `XX` with a number starting from `0`):
```
Comment (howardwu) [inline README.md]:
```suggestion
The following is a list of options to initialize a node (replace `<NODE_ID>` with a number starting from `0`):
```

File: cli/src/commands/start.rs [hunk 2]
```diff
@@ -40,50 +40,53 @@ const RECOMMENDED_MIN_NOFILES_LIMIT_VALIDATOR: u64 = 1024;
 /// Starts the snarkOS node.
 #[derive(Clone, Debug, Parser)]
 pub struct Start {
-    /// Specify the network of this node.
-    #[clap(default_value = "3", long = "network")]
+    /// The network this node should operate on [default: 3]
```
Comment (howardwu) [inline cli/src/commands/start.rs]:
```suggestion
    /// Specify the network ID of this node.
```

File: cli/src/commands/start.rs [hunk 3]
```diff
@@ -40,50 +40,53 @@ const RECOMMENDED_MIN_NOFILES_LIMIT_VALIDATOR: u64 = 1024;
...
 #[derive(Clone, Debug, Parser)]
 pub struct Start {
-    /// Specify the network of this node.
-    #[clap(default_value = "3", long = "network")]
+    /// Specify the network ID of this node.
+    #[clap(default_value = "3", long = "network", value_name = "NETWORK_ID")]
     pub network: u16,
-    /// Enables development mode, specify a unique ID for this node.
-    #[clap(long)]
+    /// Enables development mode, specify a unique ID for this node
```
Comment (howardwu) [inline cli/src/commands/start.rs]:
```suggestion
    /// Enables development mode, specify a unique ID for this node.
```

File: cli/src/commands/start.rs [hunk 4]
```diff
@@ -40,50 +40,53 @@ const RECOMMENDED_MIN_NOFILES_LIMIT_VALIDATOR: u64 = 1024;
...
-    /// Specify this as a prover, with the given account private key for this node.
-    #[clap(long = "prover")]
-    pub prover: Option<String>,
-    /// Specify this as a client, with an optional account private key for this node.
-    #[clap(long = "client")]
-    pub client: Option<String>,
-
-    /// Specify the IP address and port of a peer to connect to.
-    #[clap(default_value = "", long = "connect")]
+    /// Specify this node to run as a beacon
```
Comment (howardwu) [inline cli/src/commands/start.rs]:
```suggestion
    /// Specify this node to run as a beacon.
```

File: cli/src/commands/start.rs [hunk 5]
```diff
@@ -40,50 +40,53 @@ const RECOMMENDED_MIN_NOFILES_LIMIT_VALIDATOR: u64 = 1024;
...
-    /// Specify this as a client, with an optional account private key for this node.
-    #[clap(long = "client")]
-    pub client: Option<String>,
-
-    /// Specify the IP address and port of a peer to connect to.
-    #[clap(default_value = "", long = "connect")]
+    /// Specify this node to run as a beacon.
+    #[clap(long = "beacon", conflicts_with_all = &["client", "prover", "validator"])]
+    pub beacon: bool,
+    /// Specify this node to run as a validator
```
Comment (howardwu) [inline cli/src/commands/start.rs]:
```suggestion
    /// Specify this node to run as a validator.
```

**Comment by howardwu** [comment]
Changes the CLI in ways that were not discussed or approved.

Closing #2263 in favor of #2262.
